test/e2e_node/device_plugin.go

```diff
@@ -69,33 +69,41 @@ var _ = framework.KubeDescribe("Device Plugin [Feature:DevicePlugin] [Serial]",
 
 			By("Waiting for the resource exported by the stub device plugin to become available on the local node")
 			devsLen := int64(len(devs))
-			Eventually(func() int64 {
+			Eventually(func() bool {
 				node, err := f.ClientSet.CoreV1().Nodes().Get(framework.TestContext.NodeName, metav1.GetOptions{})
 				framework.ExpectNoError(err)
-				return numberOfDevices(node, resourceName)
-			}, 30*time.Second, framework.Poll).Should(Equal(devsLen))
+				return numberOfDevicesCapacity(node, resourceName) == devsLen &&
+					numberOfDevicesAllocatable(node, resourceName) == devsLen
+			}, 30*time.Second, framework.Poll).Should(BeTrue())
 
 			By("Creating one pod on node with at least one fake-device")
 			podRECMD := "devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs"
 			pod1 := f.PodClient().CreateSync(makeBusyboxPod(resourceName, podRECMD))
 			deviceIDRE := "stub devices: (Dev-[0-9]+)"
-			count1, devId1 := parseLogFromNRuns(f, pod1.Name, pod1.Name, 0, deviceIDRE)
+			devId1 := parseLog(f, pod1.Name, pod1.Name, deviceIDRE)
 			Expect(devId1).To(Not(Equal("")))
 
 			pod1, err = f.PodClient().Get(pod1.Name, metav1.GetOptions{})
 			framework.ExpectNoError(err)
 
-			By("Restarting Kubelet and waiting for the current running pod to restart")
+			ensurePodContainerRestart(f, pod1.Name, pod1.Name)
+
+			By("Confirming that device assignment persists even after container restart")
+			devIdAfterRestart := parseLog(f, pod1.Name, pod1.Name, deviceIDRE)
+			Expect(devIdAfterRestart).To(Equal(devId1))
+
+			By("Restarting Kubelet")
 			restartKubelet()
 
-			By("Confirming that after a kubelet and pod restart, fake-device assignement is kept")
-			count1, devIdRestart1 := parseLogFromNRuns(f, pod1.Name, pod1.Name, count1+1, deviceIDRE)
+			ensurePodContainerRestart(f, pod1.Name, pod1.Name)
```

Comment:
It might be good to check that the restart count is greater than 0

Reply:
@RenaudWasTaken https://github.com/kubernetes/kubernetes/pull/63130/files#diff-fddcd07d6f1c92a33bbe2c1cf91d59f7R237 , test will fail if restart count is not greater than 0.

Reply:
Ah cool!